packages/lwc-language-server/src/lwc-server.ts

```diff
@@ -175,7 +175,18 @@ export default class Server {
     }
 
     private shouldProvideBindingsInHTML(params: CompletionParams): boolean {
-        return params.context?.triggerCharacter === '{';
+        const position = params.position;
+        const doc = this.documents.get(params.textDocument.uri);
+        const offset = doc.offsetAt(position);
+        const text = doc.getText();
+        let startIndex = offset - 1;
+        let char = text.charAt(startIndex);
+        const regPattern = /(\w|\$)/; // Valid variable names in JavaScript can contain letters, digits, underscore or $
```

Comment:
Is it possible to add additional tests to check for these patterns?

Reply:
@randi274 Tests have been updated as well.